### PR TITLE
Upgrade Rust to 1.60.0

### PR DESCRIPTION
**Pull request by soareschen**
## Description

Updating Rust to [v1.60.0](https://blog.rust-lang.org/2022/04/07/Rust-1.60.0.html) so that the result on CI matches the result on local machine.

PSA: When updating the code base with this new `rust-toolchain.toml` file, make sure that you stop Rust Analyzer and the running editor first before pulling the new change. Otherwise you may encounter error message like:

```
the 'cargo' binary, normally provided by the 'cargo' component, is not applicable to the '1.60.0-x86_64-unknown-linux-gnu' toolchain
```

To fix that, you have to first quit Rust Analyzer and run the following to re-add the component in Rustup:

```bash
rustup component remove cargo
rustup component add cargo
```

______

### PR author checklist:
- [ ] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests: integration (for Hermes) or unit/mock tests (for modules). 
- [ ] Linked to GitHub issue.
- [ ] Updated code comments and documentation (e.g., `docs/`).

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

**Review by adizere — APPROVED** [review]

**Comment by mzabaluev** [comment]
Another PSA: whenever you want to update the MSRV (which this PR does not do), be sure to follow the checklist in the corresponding [issue template](https://github.com/informalsystems/ibc-rs/blob/master/.github/ISSUE_TEMPLATE/rust-update.md)

**Comment by soareschen** [comment]
> Another PSA: whenever you want to update the MSRV (which this PR does not do), be sure to follow the checklist in the corresponding [issue template](https://github.com/informalsystems/ibc-rs/blob/master/.github/ISSUE_TEMPLATE/rust-update.md)

Updated the Rust MSRV as well.

**Comment by romac** [comment]
Let's wait until Monday to merge this to give upstream projects a chance to catch up as well.

**Pull request merged**
